src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
         /// <param name="matrix">The source matrix to invert.</param>
         /// <param name="result">If successful, contains the inverted matrix.</param>
         /// <returns>True if the source matrix could be inverted; False otherwise.</returns>
-        public static bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
         {
+            if (Sse.IsSupported)
+            {
+                // This implementation is based on the DirectX Math Library XMMInverse method
+                // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathMatrix.inl
+
+                // Load the matrix values into rows
+                Vector128<float> row1 = Sse.LoadVector128(&matrix.M11);
+                Vector128<float> row2 = Sse.LoadVector128(&matrix.M21);
+                Vector128<float> row3 = Sse.LoadVector128(&matrix.M31);
+                Vector128<float> row4 = Sse.LoadVector128(&matrix.M41);
+
+                // Transpose the matrix
+                Vector128<float> vTemp1 = Sse.Shuffle(row1, row2, 0x44);
```

Comment:
nit: For the various shuffle constants, it would be beneficial to annotate them like:
```csharp
Sse.Shuffle(row1, row2, 0x44); // _MM_SHUFFLE(1, 0, 1, 0);
```

This allows, at a glance, for the arrangement to be determined. Alternatively, it could be done like:
```csharp
Sse.Shuffle(row1, row2, 0b_01_00_01_00);
```
Which is a bit easier to separate out/update the arrangements for (but not as clear as the comment).

Reply:
The first one is definitely more readable I think.  I added the comments to all Shuffles and Permutes.